### PR TITLE
content: correct edit links for site github repo

### PR DESCRIPTION
**Pull request by deadprogram**
Corrects https://github.com/tinygo-org/tinygo-site/issues/414 by setting the repository links per section. See https://www.docsy.dev/docs/adding-content/repository-links/

### CODE REVIEW

**Comment by nickaxgit** [comment]
subscribed, will make my edit when the merge is done. I found the device examples super useful, but they were very buried - just want to highlight them for future travellers

File: content/docs/_index.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ weight: 20
 menu:
   main:
     weight: 20
+
+cascade:
+  github_repo: https://github.com/tinygo-org/tinygo-site
+  github_subdir: content/docs
+  path_base_for_github_subdir: content/docs
+  github_branch: release
```
Comment (aykevl) [inline content/docs/_index.md]: Shouldn't this be the `dev` branch? (To avoid having to mess with the git history on the next update).

**Review by aykevl — COMMENTED** [review]
The change looks good, but shouldn't we be directing people to the dev branch?

File: content/docs/_index.md (hunk 1 above)
Comment (deadprogram) [reply content/docs/_index.md]: ok yes. good point. corrected.

**Comment by deadprogram** [comment]
Corrected branch, now merging. Thanks for review @aykevl

**Pull request merged**